core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+
+          schemaDetails.add(schemaMap);
+          topicOverview.setSchemaExists(true);
+          log.info("Getting schema " + topicNameSearch);
+        } catch (Exception e) {
+          log.error("Error ", e);
+        }
+      }
+      if (topicOverview.isSchemaExists()) topicOverview.setSchemaDetails(schemaDetails);
+    }
+  }
+
+  private void updateTopicOverviewItems(
+      String topicNameSearch,
+      int tenantId,
+      Integer loggedInUserTeam,
+      List<Topic> topics,
+      TopicOverview topicOverview,
+      List<TopicInfo> topicInfoList,
+      List<AclInfo> aclInfo,
+      Integer topicOwnerTeam) {
+    try {
+      if (Objects.equals(topicOwnerTeam, loggedInUserTeam)) {
+        topicOverview.setPromotionDetails(getTopicPromotionEnv(topicNameSearch, topics, tenantId));
+
+        if (topicInfoList.size() > 0) {
+          TopicInfo lastItem = topicInfoList.get(topicInfoList.size() - 1);
+          lastItem.setTopicDeletable(
+              aclInfo.stream()
+                  .noneMatch(
+                      aclItem -> Objects.equals(aclItem.getEnvironment(), lastItem.getCluster())));
+          lastItem.setShowDeleteTopic(true);
+        }
+      } else {
+        Map<String, String> hashMap = new HashMap<>();
+        hashMap.put("status", "not_authorized");
+        topicOverview.setPromotionDetails(hashMap);
+      }
+    } catch (Exception e) {
+      Map<String, String> hashMap = new HashMap<>();
+      hashMap.put("status", "not_authorized");
+      topicOverview.setPromotionDetails(hashMap);
+    }
+  }
+
+  private List<AclInfo> getAclInfoList(
+      int tenantId,
+      TopicOverview topicOverview,
+      List<TopicInfo> topicInfoList,
+      List<AclInfo> aclInfo,
+      List<AclInfo> prefixedAclsInfo) {
+    aclInfo = aclInfo.stream().distinct().collect(Collectors.toList());
+    List<AclInfo> transactionalAcls =
+        aclInfo.stream()
+            .filter(aclRec -> aclRec.getTransactionalId() != null)
+            .collect(Collectors.toList());
+
+    for (AclInfo aclInfo1 : aclInfo) {
+      aclInfo1.setEnvironmentName(getEnvDetails(aclInfo1.getEnvironment(), tenantId).getName());
+    }
+
+    for (AclInfo aclInfo2 : prefixedAclsInfo) {
+      aclInfo2.setEnvironmentName(getEnvDetails(aclInfo2.getEnvironment(), tenantId).getName());
+    }
+
+    topicOverview.setAclInfoList(aclInfo);
+    if (prefixedAclsInfo.size() > 0) {
+      topicOverview.setPrefixedAclInfoList(prefixedAclsInfo);
+      topicOverview.setPrefixAclsExists(true);
+    }
+    if (transactionalAcls.size() > 0) {
+      topicOverview.setTransactionalAclInfoList(transactionalAcls);
+      topicOverview.setTxnAclsExists(true);
+    }
+
+    topicOverview.setTopicInfoList(topicInfoList);
+    return aclInfo;
+  }
+
+  private void enrichTopicInfoList(
+      String topicNameSearch,
+      HandleDbRequests handleDb,
+      int tenantId,
+      Integer loggedInUserTeam,
+      ArrayList<String> reqTopicsEnvsList,
```

Comment:
There is only check for contains among usages of this parameter.
So it would make sense to use `Set` rather than `List`